subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -332,7 +332,8 @@ public NodeInVisitingSegment apply(Node node) {
         }
         executionQueue.clear();
         dependencyResolver.clear();
-        Iterables.addAll(executionQueue, nodeMapping);
+        nodeMapping.removeIf(Node::requiresMonitoring);
```

Comment:
Foreign nodes require monitoring for completion and should not be executed in the current build.
Previously they were not included here as they did not participate in the task graph construction.